src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java

```diff
@@ -544,6 +590,24 @@ OAUTH2_ACCESS_TOKEN_URL_CONFIG, getString(OAUTH2_ACCESS_TOKEN_URL_CONFIG),
 
     }
 
+    private void handleMissingConfigurationForOAuth2() {
+        Stream.of(OAUTH2_ACCESS_TOKEN_URL_CONFIG, OAUTH2_GRANT_TYPE_PROP_CONFIG, OAUTH2_GRANT_TYPE_CONFIG,
+                  OAUTH2_CLIENT_ID_PROP_CONFIG, OAUTH2_CLIENT_ID_CONFIG, OAUTH2_CLIENT_SECRET_PROP_CONFIG)
+              .filter(configKey -> getString(configKey) == null || getString(configKey).isBlank())
+              .findFirst()
+              .ifPresent(missingConfiguration -> {
+                  throw new ConfigException(missingConfiguration, getString(missingConfiguration),
+                          "Must be present when " + HTTP_HEADERS_AUTHORIZATION_CONFIG + " = "
+                          +
+                          AuthorizationType.OAUTH2);
+              });
+
+        if (oauth2ClientSecret() == null || oauth2ClientSecret().value().isEmpty()) {
+            throw new ConfigException(OAUTH2_CLIENT_SECRET_CONFIG, oauth2ClientSecret(),
+                    "Must be present when " + HTTP_HEADERS_AUTHORIZATION_CONFIG + " = " + AuthorizationType.OAUTH2);
+        }
+    }
```

Comment:
If I'm reading this correctly: From `...RequestForm` builder, when a form is built, it will fail if both `clientId` and `clientSecret` are not set together.
As both of this values are configured here, we could fail faster when connector is configured instead of deploy connector and wait for the first message to fail -- therefore the suggestion to add the validation also here. Am I missing something?